tests/utils/test_decorators.py

```diff
@@ -0,0 +1,182 @@
+"""Tests for decorator utilities."""
+
+import os
+
+import pytest
+
+from mcp.server.fastmcp.exceptions import ToolError
+
+from linux_mcp_server.utils.decorators import disallow_local_execution_in_containers
+
+
+@pytest.fixture
+def basic_test_func():
+    """Fixture for basic test function with host and username parameters."""
+
+    @disallow_local_execution_in_containers
+    async def test_func(host=None, username=None):
+        return "success"
+
+    return test_func
+
+
+@pytest.fixture
+def parametric_test_func():
+    """Fixture for test function that returns formatted parameters."""
+
+    @disallow_local_execution_in_containers
+    async def test_func(host=None, username=None):
+        return f"host={host}, user={username}"
+
+    return test_func
+
+
+@pytest.fixture
+def host_only_test_func():
+    """Fixture for test function with only host parameter."""
+
+    @disallow_local_execution_in_containers
+    async def test_func(host=None):
+        return "success"
+
+    return test_func
+
+
+class TestDisallowLocalExecutionInContainers:
+    """Test disallow_local_execution_in_containers decorator."""
+
+    async def test_allows_execution_when_host_provided(self, basic_test_func):
+        """Test that execution is allowed when host parameter is provided."""
+        # Should not raise when host is provided
+        result = await basic_test_func(host="remote.example.com", username="user")
+        assert result == "success"
+
+    async def test_allows_execution_when_not_in_container(self, basic_test_func, mocker):
+        """Test that local execution is allowed when not running in a container."""
+        # Ensure we're not in container environment
+        mocker.patch.dict(os.environ, {}, clear=False)
+        if "container" in os.environ:
+            del os.environ["container"]
+        result = await basic_test_func(host=None, username="user")
+        assert result == "success"
+
+    async def test_raises_error_when_local_in_container(self, basic_test_func, mocker):
+        """Test that ToolError is raised when attempting local execution in a container."""
+        # Simulate running in a container
+        mocker.patch.dict(os.environ, {"container": "docker"})
```

Comment:
Monkeypatch could also be used here. There's no real functional difference, though.

```suggestion
        monkeypatch.setenv("container", "docker")
```